File: Java-Core/src/main/java/presto/error/DivideByZeroError.java
package presto.error;

import presto.expression.IExpression;
import presto.grammar.CategorySymbol;
import presto.grammar.Identifier;
import presto.runtime.Context;


public class DivideByZeroError extends ExecutionError {

	private static final long serialVersionUID = 1L;

	@Override
	public IExpression getExpression(Context context) {
		return context.getRegisteredValue(CategorySymbol.class, new Identifier("DIVIDE_BY_ZERO"));
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/error/DivideByZeroError.js
var ExecutionError = require("./ExecutionError").ExecutionError;

function DivideByZeroError() {
	ExecutionError.call(this);
	return this;
}

DivideByZeroError.prototype = Object.create(ExecutionError.prototype);
DivideByZeroError.prototype.constructor = DivideByZeroError;

DivideByZeroError.prototype.getExpression = function(context) {
	return context.getRegisteredValue("DIVIDE_BY_ZERO");
};

exports.DivideByZeroError = DivideByZeroError;
